src/components/DappBrowser/handleProviderRequest.ts

```diff
@@ -344,9 +343,9 @@ export const handleProviderRequestApp = ({ messenger, data, meta }: { messenger:
     checkRateLimit,
     onSwitchEthereumChainNotSupported,
     onSwitchEthereumChainSupported,
-    getProvider: chainId => getProvider({ chainId: chainId as number }) as unknown as Provider,
+    getProvider: ({ chainId }) => getProvider({ chainId: chainId as number }),
     getActiveSession,
-    getChain: chainId => defaultChains[chainId] as Chain,
+    getChainNativeCurrency: chainId => chainsNativeAsset[chainId],
```

Comment:
this is not part of the "main fix" but is a consequence of updating to the latest provider package